pkg/webhook/server.go

```diff
@@ -0,0 +1,202 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webhook
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"path"
+	"sync"
+	"time"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	apitypes "k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/cert"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/cert/writer"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/types"
+)
+
+// ServerOptions are options for configuring an admission webhook server.
+type ServerOptions struct {
+	// KVMap can be used to pass in any arbitrary information that the admission.Handler needs.
```

Comment:
I think it would be worth documenting that this information will turn up as context values given to the Handler.